File: Day26/SearchRangeSortedArray.java
import java.util.*;
class SearchRangeSortedArray {
    public int[] searchRange(int[] nums, int target) {
        int[] result = {-1, -1};
        int left = 0, right = nums.length - 1;

        while (left <= right) {
            int mid = (left + right) / 2;
            if (nums[mid] == target) {
                result[0] = mid;
                right = mid - 1;
            } else if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        left = 0; right = nums.length - 1;

        while (left <= right) {
            int mid = (left + right) / 2;
            if (nums[mid] == target) {
                result[1] = mid;
                left = mid + 1;
            } else if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return result;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter array size:");
        int n = sc.nextInt();
        int[] nums = new int[n];

        System.out.println("Enter sorted array elements:");
        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();
        }

        System.out.println("Enter target:");
        int target = sc.nextInt();

        SearchRangeSortedArray sol = new SearchRangeSortedArray();
        int[] res = sol.searchRange(nums, target);
        System.out.println("First and Last Position: " + Arrays.toString(res));
    }
}

File: Day26/SearchRangeSortedArray.py
from typing import List

class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def findIndex(isFirst):
            left, right = 0, len(nums) - 1
            idx = -1
            while left <= right:
                mid = (left + right) // 2
                if nums[mid] == target:
                    idx = mid
                    if isFirst:
                        right = mid - 1
                    else:
                        left = mid + 1
                elif nums[mid] < target:
                    left = mid + 1
                else:
                    right = mid - 1
            return idx
        start = findIndex(True)
        end = findIndex(False)
        return [start, end]

if __name__ == "__main__":
    n = int(input("Enter number of elements: "))
    nums = list(map(int, input("Enter sorted array elements: ").split()))
    target = int(input("Enter target value: "))
    sol = Solution()
    result = sol.searchRange(nums, target)
    print("First and Last Position:", result)
